interface/src/error.rs

```diff
@@ -274,6 +275,83 @@ impl From<TokenError> for ProgramError {
+            16 => Ok(Self::MintCannotFreeze),
+            17 => Ok(Self::AccountFrozen),
+            18 => Ok(Self::MintDecimalsMismatch),
+            19 => Ok(Self::NonNativeNotSupported),
+            20 => Ok(Self::ExtensionTypeMismatch),
+            21 => Ok(Self::ExtensionBaseMismatch),
+            22 => Ok(Self::ExtensionAlreadyInitialized),
+            23 => Ok(Self::ConfidentialTransferAccountHasBalance),
+            24 => Ok(Self::ConfidentialTransferAccountNotApproved),
+            25 => Ok(Self::ConfidentialTransferDepositsAndTransfersDisabled),
+            26 => Ok(Self::ConfidentialTransferElGamalPubkeyMismatch),
+            27 => Ok(Self::ConfidentialTransferBalanceMismatch),
+            28 => Ok(Self::MintHasSupply),
+            29 => Ok(Self::NoAuthorityExists),
+            30 => Ok(Self::TransferFeeExceedsMaximum),
+            31 => Ok(Self::MintRequiredForTransfer),
+            32 => Ok(Self::FeeMismatch),
+            33 => Ok(Self::FeeParametersMismatch),
+            34 => Ok(Self::ImmutableOwner),
+            35 => Ok(Self::AccountHasWithheldTransferFees),
+            36 => Ok(Self::NoMemo),
+            37 => Ok(Self::NonTransferable),
+            38 => Ok(Self::NonTransferableNeedsImmutableOwnership),
+            39 => Ok(Self::MaximumPendingBalanceCreditCounterExceeded),
+            40 => Ok(Self::MaximumDepositAmountExceeded),
+            41 => Ok(Self::CpiGuardSettingsLocked),
+            42 => Ok(Self::CpiGuardTransferBlocked),
+            43 => Ok(Self::CpiGuardBurnBlocked),
+            44 => Ok(Self::CpiGuardCloseAccountBlocked),
+            45 => Ok(Self::CpiGuardApproveBlocked),
+            46 => Ok(Self::CpiGuardSetAuthorityBlocked),
+            47 => Ok(Self::CpiGuardOwnerChangeBlocked),
+            48 => Ok(Self::ExtensionNotFound),
+            49 => Ok(Self::NonConfidentialTransfersDisabled),
+            50 => Ok(Self::ConfidentialTransferFeeAccountHasWithheldFee),
+            51 => Ok(Self::InvalidExtensionCombination),
+            52 => Ok(Self::InvalidLengthForAlloc),
+            53 => Ok(Self::AccountDecryption),
+            54 => Ok(Self::ProofGeneration),
+            55 => Ok(Self::InvalidProofInstructionOffset),
+            56 => Ok(Self::HarvestToMintDisabled),
+            57 => Ok(Self::SplitProofContextStateAccountsNotSupported),
+            58 => Ok(Self::NotEnoughProofContextStateAccounts),
+            59 => Ok(Self::MalformedCiphertext),
+            60 => Ok(Self::CiphertextArithmeticFailed),
+            61 => Ok(Self::PedersenCommitmentMismatch),
+            62 => Ok(Self::RangeProofLengthMismatch),
+            63 => Ok(Self::IllegalBitLength),
+            64 => Ok(Self::FeeCalculation),
+            65 => Ok(Self::IllegalMintBurnConversion),
+            66 => Ok(Self::InvalidScale),
+            67 => Ok(Self::MintPaused),
+            68 => Ok(Self::PendingBalanceNonZero),
+            _ => Err(ProgramError::InvalidArgument),
+        }
+    }
```

Comment:
We can avoid the huge match section (and make it future-proof) by using `#[repr(u32)]` and taking advantage of `num_derive::FromPrimitive`, which is already derived and provides `from_u32() -> Option`

```suggestion
    #[inline]
    fn try_from(code: u32) -> Result<Self, Self::Error> {
        num_traits::FromPrimitive::from_u32(code).ok_or(ProgramError::InvalidArgument)
    }
```